File: my-project (2)/my-project/my-project/src/Services/MessageService.ts
import axios from "axios";

const API_URL = "https://jobhook-46y3.onrender.com";

const user = JSON.parse(
  localStorage.getItem("user") || "{}"
);

const token = user.jwt;

const config = {
  headers: {
    Authorization: `Bearer ${token}`
  }
};

export const sendMessage = async (data:any) => {

  const res = await axios.post(
    `${API_URL}/messages/send`,
    data,
    config
  );

  return res.data;
};

export const getConversation = async (
  senderId:number,
  receiverId:number
) => {

  const res = await axios.get(
    `${API_URL}/messages/${senderId}/${receiverId}`,
    config
  );

  return res.data;
};

export const getAllUserMessages = async (
  profileId:number
) => {

  const res = await axios.get(
    `${API_URL}/messages/user/${profileId}`
  );

  return res.data;
};

File: Job-Portal/Job-Portal-20260422T161850Z-3-001/Job-Portal/src/main/java/com/jobportal/service/MessageService.java
package com.jobportal.service;

import com.jobportal.dto.MessageDTO;
import com.jobportal.exception.JobPortalException;


import java.util.List;


public interface MessageService {

    MessageDTO sendMessage(MessageDTO dto)
            throws JobPortalException;

    List<MessageDTO> getConversation(
            Long senderId,
            Long receiverId
    );

    List<MessageDTO> getAllUserMessages(
            Long profileId
    );
}